src/NuGetForUnity/Editor/FileSystemHelper.cs

```diff
@@ -0,0 +1,236 @@
+﻿using System;
+using System.IO;
+using UnityEngine;
+
+namespace NugetForUnity
+{
+    /// <summary>
+    ///     Helper class for file system operations.
+    /// </summary>
+    internal static class FileSystemHelper
+    {
+        /// <summary>
+        ///     Recursively moves all files and sub-directories from one directory to another.
+        ///     If a file or directory already exists inside the target directory, it will be overwritten.
+        /// </summary>
+        /// <param name="sourceDirectoryPath">The path to the folder to move from.</param>
+        /// <param name="destDirectoryPath">The path to the folder to move to.</param>
+        internal static void DirectoryMove(string sourceDirectoryPath, string destDirectoryPath)
+        {
+            if (!Directory.Exists(destDirectoryPath))
+            {
+                // target doesn't exist so we can simply move the folder
+                NugetLogger.LogVerbose("Moving {0} to {1}", sourceDirectoryPath, destDirectoryPath);
+                Directory.Move(sourceDirectoryPath, destDirectoryPath);
+                return;
+            }
+
+            // move the files
+            var files = Directory.GetFiles(sourceDirectoryPath);
```

Comment:
I think it wouldn't work in this case as we are deleting / moving files while we iterate. I don't know exactly what happens but I think it is like with other collections that you are not allowed to remove items while iterating.

Reply:
Actually file system iterating is specifically exempt from that rule because it is just calling the system API for getting the next available file in the search path.